src/components/LanguagePicker/index.tsx

```diff
@@ -0,0 +1,294 @@
+import { useEffect, useRef, useState } from "react"
+import { useRouter } from "next/router"
+import { useTranslation } from "next-i18next"
+import {
+  Box,
+  Flex,
+  Input,
+  Menu,
+  MenuDivider,
+  MenuItem as ChakraMenuItem,
+  MenuList,
+  type MenuListProps,
+  type MenuProps,
+  Text,
+} from "@chakra-ui/react"
+
+import type { Lang, LocaleDisplayInfo } from "@/lib/types"
+
+import { Button } from "@/components/Buttons"
+import { BaseLink } from "@/components/Link"
+
+import progressData from "@/data/translationProgress.json"
+
+import { DEFAULT_LOCALE } from "@/lib/constants"
+
+import MenuItem from "./MenuItem"
+import NoResultsCallout from "./NoResultsCallout"
+
+import i18nConfig from "@/../i18n.config.json"
+
+type LanguagePickerProps = Omit<MenuListProps, "children"> & {
+  children: React.ReactNode
+  placement: MenuProps["placement"]
+  handleClose?: () => void
+}
+
+const LanguagePicker = ({
+  children,
+  placement,
+  handleClose,
+  ...props
+}: LanguagePickerProps) => {
+  const { t } = useTranslation("page-languages")
+  const { locale, locales } = useRouter()
+  const inputRef = useRef<HTMLInputElement>(null)
+  const firstItemRef = useRef<HTMLAnchorElement>(null)
+  const [filterValue, setFilterValue] = useState("")
+
+  // Get the preferred languages for the users browser
+  const [navLangs, setNavLangs] = useState<string[]>([])
+  useEffect(() => {
+    setNavLangs(Array.from(navigator.languages))
+  }, [])
+
+  if (!(progressData?.length > 0))
```

Comment:
Consider putting the rest of this logic inside a custom hook to isolate the logic and more important, wrap this logic inside a `useEffect` because we don't need to run it in every render. There are performance issues if you force it a bit.

```ts
// I'd aim to have something like this
const { browserLocalesInfo, filteredNames } = useLanguagePicker(filterValue)

// then on the useLanguagePicker.ts file
function useLanguagePicker(filterValue: string) {
  useEffect(() => {
    // perform all the filtering and mapping when the filter value change
    
    setFilteredNames(filteredNames)
    setBrowserLocalesInfo(browserLocalesInfo)
  }, [filterValue])

  return {
    filteredNames,
    browserLocalesInfo,
  }
}
```